pandas/tests/frame/methods/test_shift.py

```diff
@@ -186,10 +189,60 @@ def test_tshift(self, datetime_frame):
         tm.assert_frame_equal(unshifted, inferred_ts)
 
         no_freq = datetime_frame.iloc[[0, 5, 7], :]
-        msg = "Freq was not given and was not set in the index"
+        msg = "Freq was not set in the index hence cannot be inferred"
         with pytest.raises(ValueError, match=msg):
             no_freq.tshift()
 
+    def test_tshift_deprecated(self, datetime_frame):
+        # GH#11631
+        with tm.assert_produces_warning(FutureWarning):
+            datetime_frame.tshift()
+
+    def test_shift_with_freq(self, datetime_frame):
+        # PeriodIndex
+        ps = tm.makePeriodFrame()
+        shifted = ps.shift(1, freq="infer")
+        unshifted = shifted.shift(-1, freq="infer")
+
+        tm.assert_frame_equal(unshifted, ps)
+
+        shifted2 = ps.shift(freq="B")
+        tm.assert_frame_equal(shifted, shifted2)
+
+        shifted3 = ps.shift(freq=offsets.BDay())
```

Comment:
can you put the failures in a separate test (e.g. things that assert raises, name it like test_shift_with_freq_errors

also not averse to separating PI and DTI into 2 tests. if possibly to parameterize would be great but if not ok too.

Reply:
done